### ISSUE TITLE
DataTables: Multiple a11y issues

### ISSUE THREAD

**Issue by McManning** [osuresearch/ui#95]
Note the list of issues by number are from the PI Portal manual evaluation (internal document to ORIS). Look at that document for full details on context, WCAG citations, recommendations. Language has been adjusted a bit to be generalized for GitHub.

Not every item is serious, but overall I'd consider this to be a severe issue that warrants a replacement. 

--

Issue 9: The filter dropdown does not announce their states of expanded/collapsed upon activation.

Issue 16: NVDA announces “clickable” for many of the table elements that are not interactive components. 

Issue 21: Selectable row checkboxes cannot be activated with space or return. 

Issue 22: Ddata cells are not programmatically associated with their column header.

Issue 26: Column headers provide sorting but if the column is not already sorted, the screen reader does not indicate that functionality either with text or by announcing the visible arrows. 

Issue 30: When the text spacing test is applied, content overlaps in the table.

Issue 32: Nested table rows have a chevron icon indicating expand/collapse. These buttons do not announce any accessible name. 

Issue 33:  Where pagination is present, name, role, and value are not announced.

Issue 40: With the screen reader on, the checkboxes are not consistently checking/unchecking with a double tap, nor are they announcing the correct state consistently.

**Comment by McManning** [comment]
Also of note, we do a direct import of `primereact/datatable` (v6.6.0) as UI v4 guidance, and have included custom CSS to standardize it with our brand. *But* we do not provide any form of React component directly from UI 4 for data tables.
